plutus-core-interpreter/src/PlutusCoreInterpreter/CekMachine.hs

```diff
@@ -0,0 +1,118 @@
+-- | The CEK machine.
+-- Rules are the same as for the CK machine from "Language.PlutusCore.Evaluation.CkMachine",
+-- except we do not use substitution and use environments instead.
+-- The CEK machine relies on variables having
+-- 1. equal 'Unique's whenever they have equal string names
+-- 2. non-equal 'Unique's whenever they have non-equal string names
+-- I.e. 'Unique's are used instead of string names, so a renamer pass is required.
+-- This is for efficiency reasons.
+-- The CEK machine generates booleans along the way which might contain globally non-unique 'Unique's.
+-- This is not a problem as the CEK machines handles name capture by design.
+
+module PlutusCoreInterpreter.CekMachine
```

Comment:
Oh, and I don't like this module prefix. At least `PlutusCore.Interpreter`. Ideally with the `Language` prefix we're using elsewhere.